### ISSUE TITLE
Segfault when launching the 1.9.0 appimage

### ISSUE THREAD

**Issue by Nixuge** [localsend/localsend#409]
### Describe the bug

The app just segfaults when launching 
```
./LocalSend-1.9.0.AppImage
Gtk-Message: 17:11:08.755: Failed to load module "canberra-gtk-module"
Gtk-Message: 17:11:08.796: Failed to load module "colorreload-gtk-module"
method call InitSystemTray
[1]    15712 segmentation fault (core dumped)  ./LocalSend-1.9.0.AppImage
```

### To reproduce

Download the appimage
Launch it

### Expected behavior

A working app

### Desktop (please complete the following information)

Arch Linux, kernel Linux 6.2.12-zen1-1-zen, can provide more info about installed packages if needed

### Additional context

After testing and updating the AUR package, the 1.9.0 version built from source seems to be working just fine.

**Comment by Tienisto** [comment]
The version `1.9.0` uses a new dependency `libayatana-appindicator3-dev` to enable the tray feature: #404 

This might be related to this. Actually, I am not very satisfied with AppImages as they seem to be a blackbox to me. But I don't know if a traditional `.tar` (the output of the flutter compilation) would be better.

**Comment by CIAvash** [comment]
Also the Arch Linux package for `libayatana` seems to be `libayatana-appindicator`.

**Comment by Nixuge** [comment]
Already had libcanberra before and the AppImage throws a different error without libayatana-appindicator (`error while loading shared libraries: libayatana-appindicator3.so.1: cannot open shared object file: No such file or directory`), so already had it installed too.

`libcanberra` doesn't seem to be the issue as 1.8.0 runs just fine with the same 2 error messages (& ~15 other programs I have installed all depend on it without having any issue from what I can see), see below:
```
Gtk-Message: 17:43:20.717: Failed to load module "canberra-gtk-module"
Gtk-Message: 17:43:20.748: Failed to load module "colorreload-gtk-module"
flutter: Starting server...
flutter: Server started. (Port: 53317, HTTPS only)
```
Honestly looks like a libayatana issue.

Also @Tienisto how do you get a tar as the build result?  I'm using this to build the package
```
	flutter pub get
	flutter pub run build_runner build
	flutter build linux
```
and I only have built files uncompressed in build/linux/x64/release/bundle

**Comment by Nixuge** [comment]
"Fixed" the issue in a really dirty way, but hey at least it works. Probably won't update the localsend-bin AUR package until I find a better fix or until this is fixed in arch, but here's the way I used to fix that (it was indeed due to an `libayatana-appindicator` issue, maybe a different version in Arch?)

- Get the AppImage
- Extract it using ./LocalSend-1.9.0.AppImage --appimage-extract
- Open the `squashfs-root` folder (the AppImage's extracted root) & go to the `lib` folder inside it, keep that open for now
- Get [Debian's libayatana-appindicator deb](https://packages.debian.org/sid/libayatana-appindicator3-1)
- Extract the 3 files from Debian's deb (`libayatana-appindicator3.so`, `libayatana-appindicator3.so.1`, `libayatana-appindicator3.so.1.0.0`) 
- Move those to the `lib` folder still open from before
- Go back to the `squashfs-root` folder and just run ./AppRun

The app now runs normally

@Tienisto No idea what distro you're running but if you want to try and fix Arch's version without installing Arch, you can do the same as before but replace Debian's files by [Arch's ones](https://archlinux.org/packages/community/x86_64/libayatana-appindicator/download/) (in `/usr/lib`)

**Comment by Tienisto** [comment]
By `tar` I mean getting the raw results from `flutter build linux` and archive them manually via the command line.

I don't use Linux but I have a VirtualBox where I compile the app via Ubuntu.

Theoretically, I can try out Arch (or some Arch based distros) to investigate the problem

**Comment by Nixuge** [comment]
Ah ok, then yeah that'd honestly seem better for compatibility, but I still think AppImages are better (when they're working at least) for most people not really into technical stuff

**Comment by tesfabpel** [comment]
The libayatana error also happens with the flatpak version:
> /app/localsend/localsend_app: error while loading shared libraries: libayatana-appindicator3.so.1: cannot open shared object file: No such file or directory

Let me know if it's better to open another issue for the flatpak version.

**Comment by Tienisto** [comment]
@tesfabpel yes.

I will revert the official flatpak version to `v1.8.0` again. It seems that `automerge` just released the next version today

**Comment by Tienisto** [comment]
@TheGB0077 I will try to fix this

**Comment by Tienisto** [comment]
Well, after adding

```yaml
  include:
    - libc6:amd64
    - libayatana-appindicator3-1:amd64
```

I get

```
Gtk-Message: 02:07:39.121: Failed to load module "canberra-gtk-module"
Gtk-Message: 02:07:39.135: Failed to load module "canberra-gtk-module"
Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
```

Even adding the environment variable `XDG_DATA_DIRS` according to #212 could not fix the problem

**Comment by TheGB0077** [comment]
> Well, after adding
> 
> ```yaml
>   include:
>     - libc6:amd64
>     - libayatana-appindicator3-1:amd64
> ```
> 
> I get
> 
> ```
> Gtk-Message: 02:07:39.121: Failed to load module "canberra-gtk-module"
> Gtk-Message: 02:07:39.135: Failed to load module "canberra-gtk-module"
> Could not load a pixbuf from icon theme.
> This may indicate that pixbuf loaders or the mime database could not be found.
> ```
> 
> Even adding the environment variable `XDG_DATA_DIRS` according to #212 could not fix the problem

I just got past the latest error, I'll test it on Arch now.

**Comment by TheGB0077** [comment]
There we go! Running on Endeavour OS, Tray is working as well :)

![image](https://user-images.githubusercontent.com/39922116/234849542-26fe0622-2daf-4ead-8b05-937d092c78e9.png)

**Comment by Zeno394** [comment]
Please I request you to do something or show me how to run LocalSend on MX Linux.
I dont want to install flat pack version which is very big in size; almost 900 MB.
Thanks

**Comment by Zeno394** [comment]
when i try to run Local Send AppImage on MX Linux (./LocalSend-1.9.0.AppImage), it gives this error:


(AppImageLauncher:11505): Gtk-WARNING **: 10:47:27.594: Unable to locate theme engine in module_path: "clearlooks",

(AppImageLauncher:11505): Gtk-WARNING **: 10:47:27.595: Unable to locate theme engine in module_path: "clearlooks",
Failed to start appimagelauncherd.service: Process org.freedesktop.systemd1 exited with status 1
See user logs and 'systemctl --user status appimagelauncherd.service' for details.

(AppImageLauncher:11505): GLib-CRITICAL **: 10:47:28.048: g_file_test: assertion 'filename != NULL' failed
/tmp/.mount_LocalSHGs56f/localsend_app: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /tmp/.mount_LocalSHGs56f/lib/libsystem_tray_plugin.so)

Please show me the solution
Thanks Again

**Comment by tesfabpel** [comment]
> Please I request you to do something or show me how to run LocalSend on MX Linux. I dont want to install flat pack version which is very big in size; almost 900 MB. Thanks

The flatpak version seems to be ~55 MB when installed (21 MB download) according to the [flathub page](https://flathub.org/it/apps/org.localsend.localsend_app). The 900 MB you're seeing is probably the size of the runtime the app uses which is shared between other apps using the same runtime, so it's a one-time download.

**Comment by Zeno394** [comment]
Thanks for the explanation.

**Comment by Zeno394** [comment]
Thanks for your efforts.

When I tried to install latest version 1.9.1 on MX Linux, I got these errors.

(AppImageLauncher:5347): Gtk-WARNING **: 08:40:16.784: Unable to locate theme engine in module_path: "clearlooks",

Failed to start appimagelauncherd.service: Process org.freedesktop.systemd1 exited with status 1
See user logs and 'systemctl --user status appimagelauncherd.service' for details.

(AppImageLauncher:5347): GLib-CRITICAL **: 08:40:18.299: g_file_test: assertion 'filename != NULL' failed
/tmp/.mount_LocalS1aL0EC/localsend_app: symbol lookup error: /tmp/.mount_LocalS1aL0EC/usr/lib/x86_64-linux-gnu/libgdk-3.so.0: undefined symbol: wl_proxy_marshal_flags

Thanks again.
